### PR TITLE
✅ #161: add vitest

### PR DESCRIPTION
**Pull request by samrichca**
image

- replace chai/mocha with vitest
- move `@types` npm packages to dev dependencies
- add test coverage
- add testing to the entire workspace
- update indentation from spaces to tabs in packages/types/package.json
- fix failing tests in ClinicalProfile (missing genetic clinic property)

### CODE REVIEW

File: packages/types/tests/entities/ClinicianInvite.spec.ts [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+import { describe, expect, it } from 'vitest';
 
 import { ConsentGroup, ClinicianInvite } from '../../src/entities/index.js';
 
 describe('ClinicianInvite', () => {
 	it('Must define conditionally required fields on condition', () => {
 		expect(
 			ClinicianInvite.safeParse({
-				id: 'CVCFbeKH2Njl1G41vCQm',
+				id: 'Y5xPZvn1bvew59ViRWIf2',
```
Comment (samrichca) [inline packages/types/tests/entities/ClinicianInvite.spec.ts:28]: change IDs to nano IDs

**Review by alyssadsouza — COMMENTED** [review]
This is really nice, I like that you added the debugger config for VS Code too! As a small nitpick since we're not using `mocha` anymore, can you update all the `package.json` files to use `vitest` like you did for `consent-ui` and the root directory? It's just `packages/types` and `consent-api` that have the `test: nyc mocha` script.

Also, are we interested in reporting coverage? Looks like Vitest definitely has support for it: https://vitest.dev/guide/coverage.html

**Comment by samrichca** [comment]
replaced `nyc mocha` with `vitest` and added coverage

**Review by anncatton — COMMENTED** [review]
tried it locally, runs nicely! Glad we have something that fits in with the tests you and Alyssa have already implemented.

Couple requests:
- Can you add a more detailed PR description? I think a lot of the changes inside the types package are related to formatting, but it's not entirely clear and I want to make sure I'm not missing something
- The PR title is a "prototype" - is this still a prototype or is it intended as something we can merge in and work with right away?
- related to the above comment, the test UI code in `/dashboard`, i'm *guessing* is POC? Or maybe it was needed to have the consent-ui `test` script run without complaint? I think if it's POC i'd like to remove to avoid merging in irrelevant code we need to clean up later. Maybe there is a real UI test we can add in this PR, or could the `test` script return a mock "pass" of some sort 🤔 
- one other smol thing i noticed - is there a way to have the test script exit automatically after it's done?

**Comment by joneubank** [comment]
Thanks for sorting this out Sam.

There are changes to some of the Schemas in this PR that are not part of the scope of testing changes. Is this the consequence of this work being done on top of other changes? I would not merge those schema changes in here with the testing PR. This branch needs some cleanup to only include the relevant changes.

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
[the remark this answers is not in the document]
Comment (samrichca) [reply .vscode/launch.json:1]: I saw a comment on another PR from @joneubank saying to remove/ignore the VS code folder so reflecting on that: this file is the [default file from the Vitest docs for debugging in Node](https://vitest.dev/guide/debugging.html#vscode) so I'm going to leave it out of this PR

File: packages/types/tests/entities/ClinicalProfile.spec.ts [hunk 3]
```diff
@@ -62,6 +63,7 @@ describe('ClinicalProfile', () => {
 				clinicalProfilePrivateKey: '7yLTLyJC2LPjpy2vBnkFm',
 				familyHistoryOfCancer: HistoryOfCancer.enum.UNKNOWN,
 				gender: Gender.enum.PREFER_TO_SELF_IDENTIFY, // chose to self-identify
+				geneticsClinicVisited: GeneticsClinic.enum.HOSPITAL_FOR_SICK_CHILDREN_TORONTO,
```
Comment (samrichca) [inline packages/types/tests/entities/ClinicalProfile.spec.ts:66]: fixed failing test

File: apps/consent-ui/package.json [hunk 4]
```diff
@@ -8,12 +8,10 @@
 		"build": "NEXT_IS_BUILDING=true next build",
 		"start": "next start",
 		"lint": "eslint .",
-		"test": "nyc mocha",
+		"test": "vitest",
 		"test-runtime-vars": "NEXT_IS_BUILDING=true next build && FEATURE_FLAG=true next start"
 	},
 	"dependencies": {
-		"@types/react": "18.2.14",
-		"@types/react-dom": "18.2.6",
```
Comment (justincorrigible) [inline apps/consent-ui/package.json:16]: 👍

File: packages/types/package.json [hunk 5]
```diff
@@ -1,28 +1,28 @@
 {
-  "name": "types",
```
Comment (samrichca) [inline packages/types/package.json:2]: formatting: spaces -> tabs

**Review by alyssadsouza — COMMENTED** [review]
This looks good to me! 💯

**Review by anncatton — APPROVED** [review]
looks good to me

**Pull request merged**